### PR TITLE
ENH: Add Contributing.md

### PR DESCRIPTION
**Pull request by jcfr**

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,63 @@
+Contributing to Slicer
+======================
+
+There are many ways to contribute to Slicer, with varying levels of effort.  Do try to
+look through the documentation first if something is unclear, and let us know how we can
+do better.
+
+  * Ask a question on the [slicer-devel email list](http://massmail.spl.harvard.edu/mailman/listinfo/slicer-devel)
+  * Submit a feature request or bug, or add to the discussion on the [Slicer issue tracker](https://github.com/Slicer/Slicer/issues)
+  * Submit a [Pull Request](https://github.com/Radiomics/Slicer/pulls) to improve Slicer or its documentation
```
Comment (jcfr) [inline CONTRIBUTING.md]: Slicer

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+5. Create a [Pull Request](https://github.com/Slicer/Slicer/pulls).
+
+This corresponds to the `Fork & Pull Model` mentioned in the [GitHub flow](https://guides.github.com/introduction/flow/index.html)
+guides.
+
+If you have push access to Slicer repository, you could simply push your branch
+into the main repository and create a [Pull Request](https://github.com/Slicer/Slicer/pulls). This corresponds to the
+`Shared Repository Model` and will facilitate other developers to checkout your
+topic without having to [configure a remote](https://help.github.com/articles/configuring-a-remote-for-a-fork/).
+It will also simplify the workflow when you are _co-developing_ a branch.
```
Comment (jcfr) [inline CONTRIBUTING.md]: Remove this section

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+When submitting a PR, make sure to add a `Cc: @Slicer/developers` comment to notify Slicer
```
Comment (jcfr) [inline CONTRIBUTING.md]:
```
cc: @Slicer/community 
```

Comment (jcfr) [inline CONTRIBUTING.md]: If you to engage specific developers, you can add `cc: @<username>` to notify them of your awesome contributions.

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+comments posted by the reviewers, you may have to revisit your patches.
+
+#### How to integrate a PR ?
+
+Getting your contributions integrated is relatively straightforward, here
+is the checklist:
+
+* All tests pass
+* Consensus is reached. This usually means that at least two reviewers added a `LGTM` comment
+and a reasonable amount of time passed without anyone objecting. `LGTM` is an
```
Comment (jcfr) [inline CONTRIBUTING.md]: (e.g 1 business day)

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+acronym for _Looks Good to Me_.
+
+Next, there are two scenarios:
+* You do NOT have push access: A Slicer core developer will integrate your PR.
+* You have push access: Simply click on the "Merge pull request" button.
```
Comment (jcfr) [inline CONTRIBUTING.md]: Remove the merge button

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+
+Then, click on the "Delete branch" button that appears afterward.
```
Comment (jcfr) [inline CONTRIBUTING.md]: Remove this

File: CONTRIBUTING.md (hunk 5 above)
Comment (jcfr) [inline CONTRIBUTING.md]: Add reference to https://www.slicer.org/wiki/Slicer:git-svn#Integrating_topic_from_external_contributor

File: CONTRIBUTING.md (hunk 4 above)
Comment (fedorov) [inline CONTRIBUTING.md]: why do we need LGTM now that github supports code review? I don't see why we need it.

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+
+#### Automatic testing of pull requests
+
+Every pull request is tested automatically using CircleCI each time you push a
+commit to it. The Github UI will restrict users from merging pull requests until
+the CI build has returned with a successful result indicating that all tests have
+passed and there were no problems detected by the linter.
+
```
Comment (jcfr) [inline CONTRIBUTING.md]: Link to https://blog.kitware.com/3d-slicer-improves-testing-for-pull-requests-using-docker-and-circleci/

File: CONTRIBUTING.md (hunk 4 above)
Comment (jcfr) [inline CONTRIBUTING.md]: Good point. Will change to `at least two reviewers approved the changes or added a LGTM comment`

**Comment by agirault** [comment]
@jcfr just putting this here in case there is anything in there that you find interesting: https://gitlab.kitware.com/iMSTK/iMSTK/blob/master/CONTRIBUTING.md

**Comment by fedorov** [comment]
I won't be able to attend the hangout today, but I suggest, as discussed in https://github.com/Slicer/Slicer/pull/674, to add a clause to the contributing guidelines that when a change has a potential to break build system and extensions, tests must include not only the core application, but extensions. It would be nice when core developers anticipate such potential problems, they could alert extension developers. Could we create a separate extension developers team to simplify alerts?

File: CONTRIBUTING.md [hunk 8]
```diff
@@ -0,0 +1,118 @@
...
+5. Hopefully, reach consensus at the hangout and proceed with the agreed plan.
+
+
+*The initial version of these guidelines was established during the [winter
+ project week 2017](http://www.na-mic.org/Wiki/index.php/2017_Winter_Project_Week/UpdatingCommunityForums).*
+
+#### Benevolent dictators for life
+
+The [benevolent dictators](https://en.wikipedia.org/wiki/Benevolent_dictator_for_life) can bypass the regular
+process and integrate changes to keep the platform healthy.
```
Comment (jcfr) [inline CONTRIBUTING.md]: , integrate changes to keep the platform healthy and help interpret or address conflict related to the contribution guidelines.

File: CONTRIBUTING.md [hunk 9]
```diff
@@ -0,0 +1,118 @@
...
+#### How to integrate a PR ?
+
+Getting your contributions integrated is relatively straightforward, here
+is the checklist:
+
+* All tests pass
+* Consensus is reached. This usually means that at least two reviewers approved
+the changes (or added a `LGTM` comment) and at least one business day passed
+without anyone objecting. `LGTM` is an acronym for _Looks Good to Me_.
+
```
Comment (jcfr) [inline CONTRIBUTING.md]: * Additionally, to accommodate developers explicitly asking for more time to test the proposed changes, integration time can be delayed by few more days.

File: CONTRIBUTING.md [hunk 10]
```diff
@@ -0,0 +1,131 @@
...
+
+#### Decision-making process
+
+1. Given the topic of interest, initiate discussion on the [mailing list](http://massmail.spl.harvard.edu/mailman/listinfo/slicer-devel).
+
+2. Identify a small circle of community members that are interested to study the
+   topic in more depth.
+
+3. Take the discussion off the general list, work on the analysis of options and
```
Comment (ihnorton) [inline CONTRIBUTING.md]: 👎  I don't think there is any reason to take code discussion off-list. It's not like the mailing list is very busy, and archived discussions can be helpful to future code explorers.

Comment (ihnorton) [inline CONTRIBUTING.md]: (in fact I would go further and say that we should make the mailing list reply-all by default, but 1997 might want the flamewar topic to be returned)

Comment (lassoan) [inline CONTRIBUTING.md]: We'll have to revise this after moving to Discourse, so I would not spend time with discussing this now.

File: CONTRIBUTING.md [hunk 11]
```diff
@@ -0,0 +1,131 @@
...
+  mailing list. For example, update of ITK, Python, Qt or VTK version, change to
+  core functionality, should be carefully reviewed and integrated. Ideally, several
+  developers would test that the changes don't break extensions.
+
+#### How to integrate a PR ?
+
+Getting your contributions integrated is relatively straightforward, here
+is the checklist:
+
+* All tests pass
```
Comment (vovythevov) [inline CONTRIBUTING.md]: Should you suggest here that your addition should be tested ? Something along the lines of:  
"Ideally, contributions should include new or improved automatic testing.  For reference, see the testing [How-tos](https://www.slicer.org/wiki/Documentation/Nightly/Developers/Tutorials/Testing)".

**Comment by jcfr** [comment]
Thanks everyone for your feedback. Integrated as [25838](http://viewvc.slicer.org/viewvc.cgi/Slicer4?view=revision&revision=25838)

**Pull request closed without merging**